src/shared/components/Tabs/Tabs.tsx

```diff
@@ -1,29 +1,112 @@
-import React, { useState } from 'react'
-import { TabsGroup, Tab } from './Tabs.styles'
+import { transitions } from '@/shared/theme'
+import { throttle } from 'lodash'
+import React, { useEffect, useRef, useState } from 'react'
+import { CSSTransition } from 'react-transition-group'
+import useResizeObserver from 'use-resize-observer'
+import { BackgroundGradient, Tab, TabsGroup, TabsWrapper, TAB_WIDTH } from './Tabs.styles'
 
-type TabsProps = {
+export type TabsProps = {
   tabs: string[]
   initialIndex?: number
   onSelectTab: (idx: number) => void
 }
 const Tabs: React.FC<TabsProps> = ({ tabs, onSelectTab, initialIndex = -1 }) => {
   const [selected, setSelected] = useState(initialIndex)
+  const [center, setCenter] = useState(0)
+  const tabsRef = useRef<HTMLDivElement>(null)
+  const [shadowsVisible, setShadowsVisible] = useState({
+    left: false,
+    right: true,
+  })
+
+  useResizeObserver<HTMLDivElement>({
+    ref: tabsRef,
+    onResize: ({ width }) => {
+      if (!tabsRef.current || !width) {
+        return
+      }
+      const isScrollBarVisible = tabsRef.current.scrollWidth > width
+      if (isScrollBarVisible) {
+        setCenter(width / 3)
+      } else {
+        setCenter(0)
+      }
+    },
```

Comment:
Could you explain a bit what's happening here? I don't understand

Reply:
I guess I don't get what `center` is supposed to represent

Reply:
This variable here basically means "the desired position of the visible middle Tab" 😅 . I used that value to move scroll a bit if the user clicks on a certain tab - `scrollLeft` will change. 
Maybe I'll rename that to just `middleTabPosition`

Reply:
`middleTabPosition` certainly sounds better, we should try to make variable names as descriptive as possible. Now that I know what that is supposed to represent, why do we do `setCenter(width / 3)`?